code/modules/projectiles/projectile.dm

```diff
@@ -185,13 +185,13 @@
 
 	if(check_ricochet(A) && check_ricochet_flag(A) && ricochets < ricochets_max)
 		ricochets++
-	if(A.handle_ricochet(src))
-		on_ricochet(A)
-		ignore_source_check = TRUE
-		range = initial(range)
-		return TRUE
-	if(firer)
-		if(A == firer || (A == firer.loc && istype(A, /obj/mecha))) //cannot shoot yourself or your mech
+		if(A.handle_ricochet(src))
+			on_ricochet(A)
+			ignore_source_check = TRUE

```

Comment:
This var was added, then promptly not utilized or plugged in properly ANYWHERE. Seriously. IF YOU PORT SOMETHING MAKE SURE YOU UNDERSTAND THE SYSTEM. THIS IS ALSO WHY YOU DON'T SET `firer` TO A NON MOB TO AVOID THE BUG OF HITTING THE MOB WHO ACTUALLY FIRED. FOR PETES SAKE.